src/coreclr/vm/object.cpp

```diff
@@ -1799,6 +1817,46 @@ BOOL Nullable::UnBoxIntoArgNoGC(ArgDestination *argDest, OBJECTREF boxedVal, Met
 
 #endif // UNIX_AMD64_ABI
 
+#if defined(TARGET_LOONGARCH64)
+
+    if (argDest->IsStructPassedInRegs())
+    {
+        // We should only get here if we are unboxing a T as a Nullable<T>
+        _ASSERTE(IsNullableType(destMT));
+
+        // We better have a concrete instantiation, or our field offset asserts are not useful
+        _ASSERTE(!destMT->ContainsGenericVariables());
+
+        if (boxedVal == NULL)
+        {
+            // Logically we are doing *dest->HasValueAddr(destMT) = false;
+            // We zero out the whole structure becasue it may contain GC references
+            // and these need to be initialized to zero.   (could optimize in the non-GC case)
+            InitValueClassArg(argDest, destMT);
+        }
+        else
+        {
+            if (!IsNullableForTypeNoGC(destMT, boxedVal->GetMethodTable()))
+            {
+                // For safety's sake, also allow true nullables to be unboxed normally.
+                // This should not happen normally, but we want to be robust
+                if (destMT == boxedVal->GetMethodTable())
+                {
+                    CopyValueClassArg(argDest, boxedVal->GetData(), destMT, 0);
+                    //if (*(UINT64*)(argDest->GetStructGenRegDestinationAddress()) == 0)
+                    //    *(UINT64*)(argDest->GetStructGenRegDestinationAddress()) = 1;
```

Comment:
```suggestion
```
Delete this if is not useful

Reply:
OK, Thanks.
I will delete it next pushing.